Adding Conda/Mamba setup and test with cloned repository notebook to docs

aditya-uw opened the pull request:
Started with a rough draft (need to fix formatting)
Including a miniforge install section, setting up OOI data access credentials, cloning the OOI-data-explorations repo, and running the test notebook.

I do also have a note on the top that some of the instructions will be repeats from the OOI-JupyterHub setup document just in case people wonder why they appear so similar.

Need to run this setup on Windows computer; will do that Friday.

Will also edit _toc.yml when I am more clear on the formatting.

leewujung commented:
@aditya-uw : Could you change this page according to the following?
- change the title of this PR
- make it a separate page called "Local environment setup"
- include it in `_toc.yml`
- the page should include the below components:
  - conda:
    - install conda/mamba using miniforge: add detailed steps
    - include windows-specific steps (have the same detailed instructions as the page you pointed me to in our page in case that one disappears, but acknowledge the source)
    - create an environment called `boat-fundamentals` using what is in `requirements.txt` (**don't** create a new env.yml file)
  - git:
    - install git
    - git clone the repository
    - navigate to 1 notebook to make sure can run the widget

Make sure you're thorough. Let me know if you have questions, and request my review when you're done. Thanks!

leewujung commented:
@aditya-uw : I agree that we don't need the pip install part with conda. PR title looks good!

aditya-uw commented:
@leewujung I have a proposed revised outline for this documentation is:
- Install Git 
  - Explain Git and GitHub
  - macOS & Linux install (explain what terminal is and why there is no need to install git)
  - Windows install (show how to install Git using screenshots)
- Install Miniforge
  - macOS & Linux install (show how to install `conda`/`mamba`)
  - Windows install (same as above but for Windows and at the end we use Git Bash to `conda init bash` and configure auto-activate for `base`)
- Testing setup
  - Using `git` to clone `fundamentals` repository (include how to fetch and pull changes with hard reset and explain participants should rename files if they want to keep their changes)
  - Create environment instructions via `conda create --name boat-fundamentals --file requirements.txt`
  - Setup environment with ipykernel
  - Navigate to notebook which I will create and run notebook
  
 I made it like this because I realized it would be best to have Git Bash set up for Windows miniforge install. And we can't create the environment if we haven't cloned the Git repository.

aditya-uw commented:
@leewujung Also could you take a look at the screenshots I've added to the "Local environment setup" page? I wanted to make sure I didn't include any sensitive information for the Git Bash portion as that was on the Windows computer.

leewujung commented:
> @leewujung, one more thing. Do we want to recommend users to create a GitHub account? I think one can actually accomplish all these steps without a GitHub account since they are only cloning and running files. Maybe I'm wrong though.

Yeah, I think it is ok to not have them creating a GitHub account. Maybe later in the summer we can expand the primer section to include the actual git materials, and there we can have them create GitHub account there.

aditya-uw commented:
@leewujung How should people open and run the notebook I made? They'll need the jupyter package so they can use `jupyter notebook` command at least so I'm wondering if I make them install with `conda`

leewujung commented:
Good point - I think we should ask them to install Jupyter Lab. Could you add that to the environment file so that have it with the conda part?

leewujung commented:
@aditya-uw : I think this is almost ready to go - thanks, great work!

I created a new section called "Computing environment" and moved all setup-related pages there, including the codespaces one. See what you think.

Could you help make all the screen grabs maybe 15-20% bigger? It looks tiny on both my laptop and external monitor. 

I still need to do a read-through (I've only been moving things around), and then we can merge.  :)

aditya-uw commented:
@leewujung 

I can do that! I think I'll leave the terminal app icon on the Install Git page alone though? I agree with the rest and I'll make them bigger and ping you again to check. I'll aim to get the text in the screengrab to be as big as the font on the page.

I'll also suggest some places I found typos to help when we revise after reading-through.

aditya-uw commented:
@leewujung I think this PR is ready for review again. Changes include:

1) I added level three subheadings into `local_01_install_git.md` because I found out about using `winget` to install Git for Windows as a single command-line function:
`winget install --id Git.Git -e --source winget`
So I added a subsection for Windows install of Git for GUI install or Command-line install.

2) I added a note below the Git GUI install if users see any installation steps not presented to use default settings or follow other online tutorials (because I saw in BOAT someone was prompted with an installation step that I did not get on my Windows computer).

3) The screenshots have been made bigger with the aim of having the text inside be close to the text outside.

aditya-uw commented:
@leewujung Just to say it here, the latest edits I made were to change the default one-liner Git install into a Git install that matches the GUI install. Some of these custom settings were recommended on Git install tutorials so I thought to include into the CLI install.
The new `winget` Git install command:
`winget install --id Git.Git -e --source winget --custom '/o:Components=ext,gitlfs,assoc,assoc_sh /o:EditorOption=Nano /o:SSHOption=OpenSSH /o:CurlOption:OpenSSL /o:BashTerminalOption=MinTTY'`

leewujung commented:
Awesome! I think we are ready to merge this.
